compiler/rustc_type_ir/Cargo.toml

```diff
@@ -10,9 +10,8 @@ bitflags = "2.4.1"
 derive-where = "1.2.7"
 ena = "0.14.3"
 indexmap = "2.0.0"
-rustc-hash = "2.0.0"
 rustc_ast_ir = { path = "../rustc_ast_ir", default-features = false }
-rustc_data_structures = { path = "../rustc_data_structures", optional = true }
+rustc_data_structures = { path = "../rustc_data_structures" }
```

Comment:
rustc_data_structures doesn't compile on stable, but rustc_type_ir needs to support stable for rust-analyzer.

Reply:
Among other things it uses `#[may_dangle]` for arenas, enables the nightly feature of hashbrown, uses rustc_macros (which needs `feature(proc_macro_diagnostics, proc_macro_tracked_env)`), rustc_serialize (which needs specialization) and it uses a whole bunch more unstable features: https://github.com/rust-lang/rust/blob/24ac030a796b9e39bc9d07b6463e633df48e485e/compiler/rustc_data_structures/src/lib.rs#L15-L36